src/pages/Login/Login.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  useDeskproAppTheme,
+  useInitialisedDeskproAppClient,
+  useQueryWithClient,
+} from "@deskpro/app-sdk";
+import { Button, H1, H5, Radio, Stack } from "@deskpro/deskpro-ui";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { InputWithTitle } from "../../components/InputWithTitle/InputWithTitle";
+import { LoadingSpinnerCenter } from "../../components/LoadingSpinnerCenter/LoadingSpinnerCenter";
+import { getWorkspaces } from "../../api/api";
+import { parseJsonErrorMessage } from "../../utils/utils";
+import { StyledLink } from "../../styles";
+
+export const Login = () => {
+  const navigate = useNavigate();
+  const { theme } = useDeskproAppTheme();
+  const [workspace, setWorkspace] = useState<string | undefined>(undefined);
+  const [apiToken, setApiToken] = useState<string | undefined>();
+  const [submitted, setSubmitted] = useState<boolean>(false);
+  const [startLogin, setStartLogin] = useState<boolean>(false);
+
+  const workspaces = useQueryWithClient(
+    ["workspaces", submitted as unknown as string, apiToken as string],
+    (client) => getWorkspaces(client),
+    { enabled: !!submitted, useErrorBoundary: false }
+  );
+
+  useInitialisedDeskproAppClient((client) => {
+    client.setTitle("Toggl Login Page");
+    client.deregisterElement("menuButton");
+  }, []);
+
+  useInitialisedDeskproAppClient(
+    (client) =>
+      (async () => {
+        if (!apiToken) return;
+
+        await client.setUserState("api_token", btoa(`${apiToken}:api_token`), {
+          backend: true,
```

Comment:
does this mean that every user brings their own API key?

Reply:
yes